### ISSUE TITLE
Can not process invalid storage units, error message is: [Invalid data source `ds_0`, error message is: Communications link failure]

### ISSUE THREAD

**Issue by windson** [apache/shardingsphere#23533]
## Bug Report

Hi, I am trying to connect to the existing tables in mysql databases db0 and db1 by registering them as storage units using the following command:

```
REGISTER STORAGE UNIT ds_0 (
    HOST="127.0.0.1",
    PORT=3306,
    DB="db0",
    USER="root",
    PASSWORD="blah"
),ds_1 (
    HOST="127.0.0.1",
    PORT=3306,
    DB="db1",
    USER="root",
    PASSWORD="blah"
);
```

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?

5.3.1

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy

### Expected behavior

Add the storage units properly.

### Actual behavior

ERROR 19000 (44000): Can not process invalid storage units, error message is: [Invalid data source `ds_0`, error message is: Communications link failure, Invalid data source `ds_1`, error message is: Communications link failure]

### Reason analyze (If you can)
DistSQL is being used.

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
On my local DB:
```
mysql -h 127.0.0.1 --user=root -p
mysql>create database db0;
mysql>create database db1;
```

Create & Connect to ShardingSphere-Proxy, 
```
docker run -d -v /Users/pavankrn/Documents/tech/sspheredock/pgsphere/apache-shardingsphere-5.3.1-shardingsphere-proxy-bin/conf:/opt/shardingsphere-proxy/conf -v /Users/pavankrn/Documents/tech/sspheredock/pgsphere/apache-shardingsphere-5.3.1-shardingsphere-proxy-bin/ext-lib:/opt/shardingsphere-proxy/ext-lib -e PORT=3308 -p13308:3308 apache/shardingsphere-proxy:latest
mysql --host=127.0.0.1 --user=root -p --port=13308 sharding_db
```

On ShardingSphere-Proxy's mysql terminal

```
use sharding_db;
REGISTER STORAGE UNIT ds_0 (
    HOST="127.0.0.1",
    PORT=3306,
    DB="db0",
    USER="root",
    PASSWORD="blah"
),ds_1 (
    HOST="127.0.0.1",
    PORT=3306,
    DB="db1",
    USER="root",
    PASSWORD="blah"
);
```

**Comment by RaigorJiang** [comment]
Hi @windson ,
From the proxy container, you should not be able to access mysql with the address of 127.0.0.1.

**Comment by windson** [comment]
@RaigorJiang I see the following error when I ran `mysql --host=127.0.0.1 --user=root -p --port=3308 sharding_db`
`bash: mysql: command not found`

Looks like I need to install mysql shell in the container. 
Alternatively, what I tried that worked for me was that I changed the `HOST="127.0.0.1",` to `HOST="host.docker.internal",`.

So the distsql command to register storage unit looks like the following, worked for me.

```
REGISTER STORAGE UNIT ds_0 (
    HOST="host.docker.internal",
    PORT=3306,
    DB="db0",
    USER="root",
    PASSWORD="blah"
),ds_1 (
    HOST="host.docker.internal",
    PORT=3306,
    DB="db1",
    USER="root",
    PASSWORD="blah"
);
```
Anyhow, thanks for the input.
